alica_engine/include/engine/AlicaEngine.h

```diff
@@ -128,35 +131,12 @@ class AlicaEngine
     Logger _log;
     std::unique_ptr<IRoleAssignment> _roleAssignment;
     PlanBase _planBase;
-    // TODO: fix this, VariableSyncModule has circular dependency with engine header
-    // VariableSyncModule _variableSyncModule;
-    std::unique_ptr<VariableSyncModule> _variableSyncModule;
+    VariableSyncModule* _variableSyncModule;
```

Comment:
forget to revert it completely, because I tried to make the variableSyncModule a stack variable. Made it a unique_ptr again and added some longer comment about the issue.